ImageGallery/src/org/sleuthkit/autopsy/imagegallery/ImageGalleryController.java

```diff
@@ -712,16 +783,22 @@ public void run() {
                 progressHandle.switchToDeterminate(files.size());
 
                 updateProgress(0.0);
+                
+               taskCompletionStatus = true;
 
                 //do in transaction
                 DrawableDB.DrawableTransaction tr = taskDB.beginTransaction();
                 int workDone = 0;
                 for (final AbstractFile f : files) {
-                    if (isCancelled() || Thread.interrupted()) {
+                    if (isCancelled()) {
```

Comment:
I think that the taskCompletionStatus flag should be false when the task is cancelled